internal/log/coder.go

```diff
@@ -0,0 +1,136 @@
+package log
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"os"
+	"time"
+
+	"cdr.dev/slog"
+	"cdr.dev/slog/sloggers/sloghuman"
+	"github.com/coder/coder/v2/agent/proto"
+	"github.com/coder/coder/v2/codersdk"
+	"github.com/coder/coder/v2/codersdk/agentsdk"
+	"github.com/coder/retry"
+	"github.com/google/uuid"
+	"golang.org/x/mod/semver"
+)
+
+var (
+	rpcConnectTimeout  = 10 * time.Second
+	logSendGracePeriod = 10 * time.Second
```

Comment:
Nit: I wonder if these should be configurable; at least the timeout feels like it should be.

Reply:
I'll leave this as a follow up, we may not end up needing an extra knob.